src/GitHub.VisualStudio/Menus/BlameLink.cs

```diff
@@ -0,0 +1,42 @@
+﻿using GitHub.Services;
+using GitHub.VisualStudio.UI;
+using NullGuard;
+using System;
+using System.Windows;
+
+namespace GitHub.VisualStudio.Menus
+{
+    public class BlameLink : LinkMenuBase, IDynamicMenuHandler
+    {
+
+        public BlameLink(IGitHubServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+        }
+
+        public Guid Guid => GuidList.guidContextMenuSet;
+        public int CmdId => PkgCmdIDList.blameCommand;
+
+        public async void Activate([AllowNull]object data = null)
+        {
+            var isgithub = await IsGitHubRepo();
+            if (!isgithub)
+                return;
+
+            var link = await GenerateLink(true);
+            if (link == null)
+                return;
+            var browser = ServiceProvider.TryGetService<IVisualStudioBrowser>();
+            browser?.OpenUrl(link.ToUri());
+
+            await UsageTracker.IncrementOpenInGitHubCount();
+        }
+
+        public bool CanShow()
+        {
+            var githubRepoCheckTask = IsGitHubRepo();
+            return githubRepoCheckTask.Wait(250) ? githubRepoCheckTask.Result : false;
```

Comment:
This was from an [old bug fix](https://github.com/github/VisualStudio/pull/334#discussion-diff-65528365) where we added `wait` to avoid blocking the UI if it takes to long to find if a repo is a GitHub repo